src/main/java/seedu/address/logic/parser/AddCommandWithPopupParser.java

```diff
@@ -0,0 +1,28 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.logic.commands.AddCommandWithPopup;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses the input arguments and creates an {@code AddCommandWithPopup}.
+ */
+public class AddCommandWithPopupParser implements Parser<AddCommandWithPopup> {
+
+    @Override
+    public AddCommandWithPopup parse(String userInput) throws ParseException {
+        requireNonNull(userInput);
+        userInput = userInput.trim().toUpperCase();
+        switch (userInput) {
+        case AddCommandWithPopup.ADD_BUYER:
+            return new AddCommandWithPopup(AddCommandWithPopup.ADD_BUYER);
+        case AddCommandWithPopup.ADD_SUPPLIER:
+            return new AddCommandWithPopup(AddCommandWithPopup.ADD_SUPPLIER);
+        default:
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommandWithPopup.MESSAGE_USAGE));
+        }
+    }
```

Comment:
Do we need to distinguish between the 5 AddCommands?

Reply:
Basically the AddCommandWithPopup will only be invoked when the command word is "add".
If the command word for AddCommand without the pop-up is something like "add-b", "add-o", then AddCommandWithPopup will not be invoked.

Then for the different cases of AddCommandWithPopup, we need to distinguish which type of person is being added as the pop up window needs to know this to fill the placeholders in the display.